internal/eva/login.go

```diff
@@ -0,0 +1,50 @@
+package eva
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+const (
+	path = "/api/core/Login"
+)
+
+type LoginCredentials struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+type LoginResponse struct {
+	AuthenticationToken string `json:"AuthenticationToken"`
+}
+
+func (c *Client) Login(ctx context.Context, credentials LoginCredentials) error {
+
+	resp, err := c.Client.R().
+		SetBody(credentials).
+		Post(path)
```

Comment:
Would you mind renaming it to smth like `loginPath` or`loginEndpoint` or `loginUri` ?